go/vt/vtgate/planbuilder/insert.go

```diff
@@ -71,8 +71,8 @@ func buildInsertUnshardedPlan(ins *sqlparser.Insert, table *vindexes.Table, vsch
 		return eRoute, nil
 	case sqlparser.Values:
 		rows = insertValues
-		if hasSubquery(rows) {
-			return nil, errors.New("unsupported: subquery in insert values")
+		if !validateSubquerySamePlan(rows, eRoute, vschema) {
```

Comment:
On Slack, I was implying that you could also point the Select & Union cases above to use the new function you wrote. I think they should work correctly, right?

If so, we'll get union subqueries for inserts for free (and a few more lines of code deleted).

But this is also good otherwise.